pkg/scalers/azure_app_insights_scaler.go

```diff
@@ -65,14 +70,36 @@ func NewAzureAppInsightsScaler(config *scalersconfig.ScalerConfig) (Scaler, erro
 		return nil, fmt.Errorf("error parsing azure app insights metadata: %w", err)
 	}
 
+	// handle HTTP client timeout
+	httpClientTimeout := config.GlobalHTTPTimeout
+	httpClient := kedautil.CreateHTTPClient(httpClientTimeout, false)
+
+	creds, err := getAuthConfig(meta.azureAppInsightsInfo, config.PodIdentity)
+	if err != nil {
+		return nil, fmt.Errorf("error getting auth config: %w", err)
+	}
+
 	return &azureAppInsightsScaler{
 		metricType:  metricType,
 		metadata:    meta,
 		podIdentity: config.PodIdentity,
 		logger:      logger,
+		httpClient:  httpClient,
+		creds:       creds,
 	}, nil
 }
 
+func getAuthConfig(info azure.AppInsightsInfo, podIdentity kedav1alpha1.AuthPodIdentity) (azcore.TokenCredential, error) {
+	switch podIdentity.Provider {
+	case "", kedav1alpha1.PodIdentityProviderNone:
+		return azidentity.NewClientSecretCredential(info.TenantID, info.ClientID, info.ClientPassword, nil)
+	case kedav1alpha1.PodIdentityProviderAzureWorkload:
+		return azure.NewADWorkloadIdentityCredential(podIdentity.GetIdentityID(), podIdentity.GetIdentityTenantID())
+	default:
+		return nil, fmt.Errorf("unknown pod identity provider: %s", podIdentity.Provider)
+	}
+}
```

Comment:
Maybe an improvement for other PRs, but code like this generating the credentials is repeated sometimes in different places, we could move it to azure pkg and reuse it (not necessarily for this PR)